Automattic/jetpack
# External Media: Implementation

External Media allows users to browse and insert [free images from Pexels](https://wordpress.com/support/free-photo-library/) or [their own Google Photos library](https://wordpress.com/support/google-photos/).

This feature extends the Media Library button in block like Image, Gallery, etc, making it easy to find and convenient to use:

<img width="558" alt="81190858-a4ceeb80-8fb8-11ea-992e-5339e9e73aac" src="https://user-images.githubusercontent.com/1398304/81990272-b4bb8080-95f3-11ea-9ed0-83733f87c40b.png">

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
It's a feature new to Jetpack but existing on Dotcom.
See 47-gh-dotcom-private

#### Testing instructions
* This PR adds a new npm module, please run `yarn && yarn run build-extensions` to pull it in.
* Open the editor and insert an image block.
* Click the Select Image button and choose either of the new libraries.
* For Pexels:
  * Change the search phrase to explore how the results update.
  * Click the Load more button and make sure the content looks right and gets inserted in the right place
  * Select an image and click "Copy & Insert"
  * Make sure the image is copied into your media library and inserted properly.
* For Google Photos:
  * Navigate to https://wordpress.com/marketing and select your test site.
  * Make sure Google Photos is disconnected.
  * Back in the Editor: Select Google Photos from the button dropdown
  * Make sure it's asking you to connect to your Google account
  * After successfully connecting, select a photo album or selectet Recent Photos from the dropdown.
  * Select an image and click "Copy & Insert"
  * Make sure the image is copied into your media library and inserted properly.

Also see demo video in paYJgx-G1-p2.


#### Proposed changelog entry for your changes:

* Adds support for Pexels and Google Photos media libraries.
